pkg/roachpb/errors.go

```diff
@@ -202,41 +181,8 @@ func (e *Error) SafeFormat(s redact.SafePrinter, _ rune) {
 		return
 	}
 
-	if e.EncodedError != (errors.EncodedError{}) {
-		err := errors.DecodeError(context.Background(), e.EncodedError)
-		var iface ErrorDetailInterface
-		if errors.As(err, &iface) {
-			// Deprecated code: if there is a detail and the message produced by the detail
-			// (which gets to see the surrounding Error) is different, then use that message.
-			// What is likely the cause of that is that someone passed an updated Transaction
-			// to the Error via SetTxn, and the error detail prints that txn.
-			//
-			// TODO(tbg): change SetTxn so that instead of stashing the transaction on the
-			// Error struct, it wraps the EncodedError with an error containing the updated
-			// txn. Make GetTxn retrieve the first one it sees (overridden by UnexposedTxn
-			// while it's still around). Remove the `message(Error)` methods from ErrorDetailInterface.
-			// We also have to remove GetDetail() in the process since it doesn't understand the
-			// wrapping; instead we need a method that looks for a specific kind of detail, i.e.
-			// we basically want to use `errors.As` instead.
-			deprecatedMsg := iface.message(e)
-			if deprecatedMsg != err.Error() {
-				s.Print(deprecatedMsg)
-				return
-			}
-		}
-		s.Print(err)
```

Comment:
this line shouldn't have gotten lost. Will reinstate it and see if that fixes anything.

Reply:
It seems to fix at least anything I was able to spot check. Let's see if CI turns green.